### PR TITLE
Add token authentication for API calls.

### PR DESCRIPTION
**Pull request by lowjoel**
This is needed for the evaluator to connect to the Coursemology server.

This is using my Simple Token Authentication gem fork, because I'm waiting on gonzalo-bulnes/simple_token_authentication#208

### CODE REVIEW

**Comment by lowjoel** [comment]
@allenwq Hakiri's warning is for an older version of Rails, the new default is `:null_session` which resets the session. This is more suitable for API calls.

File: app/models/user.rb [hunk 1]
```diff
@@ -75,6 +75,16 @@ def set_next_email_as_primary
     default_email_record.update_attributes(primary: true)
   end
 
+  # Enables token authentication for this user.
+  def enable_token_authentication
```
Comment (allenwq) [inline app/models/user.rb]: @lowjoel Should we put this method in UserAuthentication concern ?

Comment (lowjoel) [inline app/models/user.rb]: makes sense. didn't think about that.

**Comment by allenwq** [comment]
@lowjoel This PR looks good, but is there any way that we can ignore Hakiri's complain, otherwise the subsequent PRs will fail ?

**Comment by lowjoel** [comment]
I'll false positive the report later, they're using an older version of Rails.

**Pull request merged**